src/index.ts

```diff
@@ -218,8 +218,13 @@ export async function build(_options: Options) {
                 const extraPatterns = Array.isArray(options.clean)
                   ? options.clean
                   : []
+                // Don't clear d.ts files here if we are generating them
+                // They will be handled by the tsup:clean rollup plugin
```

Comment:
I don't actually understand why the cleaning is done in separate places, to be honest. It might be worth clarifying here.